objects/src/batches/note_tree.rs

```diff
@@ -0,0 +1,46 @@
+use miden_crypto::{
+    hash::rpo::RpoDigest,
+    merkle::{MerkleError, SimpleSmt},
+};
+
+use crate::{notes::NoteMetadata, BATCH_OUTPUT_NOTES_TREE_DEPTH};
+
+/// Wrapper over [SimpleSmt] for batch note tree
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct BatchNoteTree(SimpleSmt<BATCH_OUTPUT_NOTES_TREE_DEPTH>);
```

Comment:
I would probably provide a bit more documentation here. For example, mention the depth of the tree, the fact that each note is stored in two adjacent leaves etc.